startups/views.py

```diff
@@ -34,75 +34,71 @@ class StartupViewSet(viewsets.ViewSet):
      """
 
     def list(self, request):
-        # Implementation of GET METHOD - ExampLE URL: /api/startups/
+        # Example URL: /api/startups/
         # Getting ALL startups logic
-        data = {
-            'message': "Hello, ALL STARTUPS PROFILE PAGE",
-            'status': status.HTTP_200_OK
-        }
-        query_data = get_query_dict(request)  # If we need to use queries like /api/startups?name=Apple
-        if query_data:
-            data.update(query_data)
-        # Should return a list!
-        return Response(data, status=status.HTTP_200_OK)
+        startups = Startup.objects.all()
+
+        # Example URL: /api/startups/?industry=test
+        # Example URL: /api/startups/?name=test
+        industry = request.query_params.get('industry')
+        name = request.query_params.get('name')
+        other_params = request.query_params.keys() - {'industry', 'name'}
+
+        if other_params:
+            return Response({"error": "Only 'industry' and 'name' parameter is allowed"}, status=status.HTTP_400_BAD_REQUEST)
+
+        if industry:
+            startups = startups.filter(industries__icontains=industry)
+            if not startups.exists():
+                return Response({"error": f"No startups found for the industry '{industry}'"},status=status.HTTP_404_NOT_FOUND)
+        if name:
+            startups = startups.filter(startup_name__icontains=name)
+            if not startups.exists():
+                return Response({"error": f"No startups found with the name '{name}'"},status=status.HTTP_404_NOT_FOUND)
```

Comment:
Extract the validation logic into a separate function that validates the query parameters passed in and potentially filters the queryset. 

As the code may grow and parameters may be added, it is better to handle this in a separate function rather than implementing it directly in the list method. This greatly improves the code's readability and extensibility.

Reply:
![afa3db4b-badd-405f-8b14-bffd364bf47f](https://github.com/Project-Stage-Academy/UA_1155_alpha/assets/112662408/61083a07-0ea6-4c77-b42c-ed64f3a083ee)

Reply:
Додала функцію

Reply:
готово